base/complex.jl

```diff
@@ -534,7 +544,17 @@ function log(z::Complex{T}) where T<:AbstractFloat
     end
     Complex(ρρ, angle(z))
 end
-log(z::Complex) = log(float(z))
+
+_log(z::Complex{<:AbstractFloat}) = log(z)
+
+function _log(z::Complex)
+    ρ = abs(z)
+    θ = angle(z)
+
+    return Complex(log(ρ), θ)
```

Comment:
Again, this can have a large relative error in the real part, e.g. `_log(1 + 1e-20im)` gives `0.0 + 1.0e-20im` whereas `log(1 + 1e-20im)` gives `5.0e-41 + 1.0e-20im`.